src/guide/receive_onchain.md

```diff
@@ -283,3 +283,51 @@ result, err := sdkServices.Refund(refundable.BitcoinAddress, destinationAddress,
 ```
 </section>
 </custom-tabs>
+
+# Calculating fees
+
+<custom-tabs category="lang">
+<div slot="title">Dart</div>
+<section>
+
+To calculate fees, you'll need to detect if channel opening fees are needed first and then calculate the fee for the amount you'll be receiving.
```

Comment:
When the amount to be received exceeds the inbound liquidity of the node, a new channel will be opened by the LSP in order for the node to receive it. This can checked by retrieving the NodeState from the SDK and comparing the inbound liquidity to the amount to be received. If the amount is greater or equal to the inbound liquidity, a new channel opening is required.

To calculate the fees for a channel being opened by the LSP: